projects/media-viewer/src/lib/store/reducers/document.reducer.ts

```diff
@@ -30,42 +28,60 @@ export function docReducer (state = initialDocumentState,
   switch (action.type) {
 
     case fromActions.SET_DOCUMENT_ID : {
-      let url = action.payload.split('/documents/');
+      const url = action.payload.split('/documents/');
       const documentId = (url.length > 1 ? url[1] : url[0]).replace('/binary', '');
       return {
         ...state,
         documentId
       }
     }
 
-    case fromActions.ADD_PAGE: {
+    case fromActions.ADD_PAGES: {
       const payload = action.payload;
-      const  numberOfPages = (state.pages.numberOfPages <= payload.pageNumber) ?
-        payload.pageNumber : state.pages.numberOfPages;
-      const styles = {
-        left: payload.div['offsetLeft'],
-        height: payload.div['offsetHeight'],
-        width: payload.div['offsetWidth']
-      };
-      const scaleRotation = {
-        scale: payload.scale,
-        rotation: payload.rotation
-      };
-      const page = {
-        numberOfPages,
-        styles,
-        scaleRotation
-      };
+      let pages = {};
+      let pageHeight;
+      let pageWidth;
+      let hasDifferentPageSize = state.hasDifferentPageSize
+      payload.forEach(page => {
+        if (pageHeight && pageWidth &&
+          (pageHeight !== page.div['offsetHeight'] || pageWidth !== page.div['offsetWidth']) &&
+          !hasDifferentPageSize) {
```

Comment:
maybe we should put !hasDifferentPageSize as the first check, so we don't bother with the others if we already know it's true